zrml/prediction-markets/src/lib.rs

```diff
@@ -2390,11 +2461,56 @@ mod pallet {
             // If the oracle reported right, return the OracleBond, otherwise slash it to
             // pay the correct reporters.
             let mut overall_imbalance = NegativeImbalanceOf::<T>::zero();
-            if report.by == market.oracle && report.outcome == resolved_outcome {
-                Self::unreserve_oracle_bond(market_id)?;
+
+            let report_by_oracle = report.by == market.oracle;
+            let is_correct = report.outcome == resolved_outcome;
+
+            // TODO: this wrapper (if let Some and !settled) can be removed
+            // TODO: if there are only markets with outsiders, who all reserved OutsiderBond
+            let outsider_bond_pending = || -> bool {
+                if let Some(bond) = &market.bonds.outsider {
+                    if !bond.is_settled {
+                        return true;
+                    }
+                }
+                false
+            };
+
+            let unreserve_outsider = || -> DispatchResult {
+                if outsider_bond_pending() {
+                    Self::unreserve_outsider_bond(market_id)?;
+                }
+                Ok(())
+            };
+
+            let slash_outsider = || -> Result<NegativeImbalanceOf<T>, DispatchError> {
+                if outsider_bond_pending() {
+                    let imbalance = Self::slash_outsider_bond(market_id, None)?;
+                    return Ok(imbalance);
+                }
+                Ok(NegativeImbalanceOf::<T>::zero())
+            };
+
+            if report_by_oracle {
+                if is_correct {
+                    Self::unreserve_oracle_bond(market_id)?;
+                } else {
+                    let negative_imbalance = Self::slash_oracle_bond(market_id, None)?;
+                    overall_imbalance.subsume(negative_imbalance);
+                }
             } else {
-                let imbalance = Self::slash_oracle_bond(market_id, None)?;
-                overall_imbalance.subsume(imbalance);
+                // outsider report
+                if is_correct {
+                    // reward outsider reporter with oracle bond
+                    let missing = Self::repatriate_oracle_bond(market_id, &report.by)?;
```

Comment:
I don't want a fail in `on_resolution` because of `repatriate_reserved_named` here. Should we ignore the result of [`repatriate`](https://github.com/paritytech/substrate/blob/911f65be7949342e12601cc3015d5c47e9339de4/frame/balances/src/lib.rs#L1024-L1041)?

Reply:
76fadc661cc78dd3fc73eaa38f5ed00401d6d325